certs/certs.go

```diff
@@ -141,29 +141,28 @@ func ValidateFinalityCertificates(verifier gpbft.Verifier, network gpbft.Network
 // any other parts of the certificate, just that the _value_ has been signed by a majority of the
 // power.
 func verifyFinalityCertificateSignature(verifier gpbft.Verifier, powerTable gpbft.PowerEntries, nn gpbft.NetworkName, cert FinalityCertificate) error {
-	totalPower := new(gpbft.StoragePower)
-	for i := range powerTable {
-		totalPower = totalPower.Add(totalPower, powerTable[i].Power)
+	scaled, totalScaled, err := powerTable.Scaled()
+	if err != nil {
+		return xerrors.Errorf("failed to scale power table: %w", err)
```

Comment:
We scale the power on-demand when validating finality certificates because finality certificates take `PowerEntries` instead of a full `PowerTable`.